### PR TITLE
various changes to make cavern work and pass tests

### PR DESCRIPTION
**Pull request by pdowler**

### CODE REVIEW

File: cavern/src/main/java/org/opencadc/cavern/files/CavernURLGenerator.java [hunk 1]
```diff
@@ -356,6 +348,16 @@ public VOSURI validateToken(String token, VOSURI targetVOSURI, Direction directi
         throw new IllegalArgumentException("Missing node URI");
     }
 
+    URL getFilesEndpoint() {
+        String hack = "https://haproxy.cadc.dao.nrc.ca/cavern/files";
+        try {
```
Comment (andamian) [inline cavern/src/main/java/org/opencadc/cavern/files/CavernURLGenerator.java:353]: Is this hack here on purpose?

**Review by andamian — APPROVED** [review]

File: cavern/src/main/java/org/opencadc/cavern/nodes/NodeUtil.java [hunk 2]
```diff
@@ -189,19 +194,20 @@ public PosixPrincipal addToCache(Subject s) {
...
     
-    private List<PosixGroup> getFromGroupCache(List<GroupURI> input)
+    private List<PosixGroup> getFromGroupCache(Collection<GroupURI> input)
             throws ResourceNotFoundException, ResourceAlreadyExistsException {
-        log.warn("getFromGidCache: " + input.size());
         List<PosixGroup> ret = new ArrayList<>(input.size());
         List<GroupURI> cacheMiss = new ArrayList<>();
         for (GroupURI g : input) {
-            PosixGroup pg = gidCache.get(g);
+            PosixGroup pg = groupCache.get(g);
```
[the remark this answers is not in the document]
Comment (pdowler) [reply cavern/src/main/java/org/opencadc/cavern/nodes/NodeUtil.java:210]: it took a more obscure usage pattern to cause it to fail, so it never worked and I only found out with the permissions tests

File: cadc-test-vos/src/main/java/org/opencadc/conformance/vos/NodesTest.java [hunk 3]
```diff
@@ -596,6 +623,26 @@ public void testLimit() {
             log.info("put: " + parentURI + " -> " + parentURL);
             ContainerNode parent = new ContainerNode(parentName);
             put(parentURL, parentURI, parent);
+            
+            if (!paginationSupported) {
```
[the remark this answers is not in the document]
Comment (pdowler) [reply cadc-test-vos/src/main/java/org/opencadc/conformance/vos/NodesTest.java:641]: yes, there are two test methods - the main thing is the immediate return and repeating the test is pointless

will reduce

File: cavern/src/main/java/org/opencadc/cavern/files/CavernURLGenerator.java (hunk 1 above)
Comment (pdowler) [reply cavern/src/main/java/org/opencadc/cavern/files/CavernURLGenerator.java:353]: just temporary to get it working... now I am at the point where I can get it dynamically. will fix

File: cavern/src/main/java/org/opencadc/cavern/nodes/NodeUtil.java [hunk 4]
```diff
@@ -355,220 +358,151 @@ public static void setPosixOwnerGroup(Path p, Integer owner, Integer group) thro
...
-                    }
+        // update action in library handles the merge
+        // clear existing
+        Map<String,String> cur = ExtendedFileAttributes.getAttributes(path);
+        for (Map.Entry<String,String> me : cur.entrySet()) {
+            ExtendedFileAttributes.setFileAttribute(path, me.getKey(), null);
+        }
+        for (NodeProperty prop : node.getProperties()) {
+            if (!FILESYSTEM_PROPS.contains(prop.getKey())) {
+                if (node instanceof LinkNode) {
```
[the remark this answers is not in the document]
Comment (pdowler) [reply cavern/src/main/java/org/opencadc/cavern/nodes/NodeUtil.java:369]: it is just extended attrs that are not supported on symbolic links; the method also sets permissions and until that is known to fail (by a NodesTest that tries to do it) it should be attempted

**Pull request merged**